File: 1kurs/zadanie23.pas
uses Crt;

type number = record
     numer:integer;
      temp:integer;
end;

var  files:array[1..10] of number;
m, i, n, s:integer;
         f:file of number;
Begin
 n:=3;
 s:=0;
 clrscr;
 assign(f, 'A.txt');
 for i:=1 to n do
 Begin
  write('Enter natural number: ');
  read(files[i].numer);
 end;
 rewrite(f);
 for i:=1 to n do
  write(f, files[i]);
 close(f);
 clrscr;
 write('Enter number m: ');
 read(m);
 assign(f, 'A.txt');
 {$I-}
 Reset(f);
 {$I+}
 if (IOResult <> 0) then
 Begin
  writeln('Error');
  exit;
 end;
 for i:=1 to filesize(f) do
 Begin
  read(f, files[i]);
  if ((files[i].numer mod m)=0) then
  Begin
   files[s+1].temp := files[i].numer;
               s := s+1;
  end;
 end;
 close(f);
 assign(f, 'B.txt');
 rewrite(f);
 for i:=1 to s do
  write(f, files[i]);
 close(f);
 assign(f, 'B.txt');
 Reset(f);
 for i:=1 to filesize(f) do
 Begin
  read(f, files[i]);
  writeln(files[i].temp);
 end;
 close(f);
end.

File: 1kurs/zadanie23.cpp
#include <stdio.h>
#include <stdlib.h>
#include <iostream>
#include <string.h>
using namespace std;

int i, m, n=3;

struct number
{
        int numer;
        float temp;
};

number files[10];
int main()
{
        system("clear");
        FILE *f;
        f = fopen("A.txt", "wb");
        for (i = 0; i<n; i++)
        {
                cout << "Enter natural number: ";
                cin  >> files[i].numer;
        }
        for (i=0; i<n; i++)
                fwrite(&files[i], sizeof(number), 1, f);
        fclose(f);
        system("clear");
        cout << "Enter number m: ";
        cin  >> m;
        f = fopen("A.txt", "r");
        fseek(f, 0, SEEK_END);
        int z = ftell(f);
        int s = 0;
        fseek(f, 0, SEEK_SET);
        for (i=0; i<z/sizeof(number); i++)
        {
                fread(&files[i], sizeof(number), 1, f);
                if ((files[i].numer % m)==0)
                {
                        files[s].temp = files[i].numer;
                        s++;
                }
        }
        fclose(f);
        f = fopen("B.txt", "wb");
        for (i=0; i<s; i++)
                fwrite(&files[i], sizeof(number), 1, f);
        fclose(f);
        f = fopen("B.txt", "r");
        fseek(f, 0, SEEK_END);
        int b = ftell(f);
        fseek(f, 0, SEEK_SET);
        for (i=0; i<s; i++)
        {
                fread(&files[i], sizeof(number), 1, f);
                cout << files[i].temp << endl;
        }
        fclose(f);
}
